docs/content/specs/shared/_index.md

```diff
@@ -249,6 +273,53 @@ To see a directory and file structure for a module, see the language specific co
 
 {{< /hint >}}
 
+##### Required Resources/Dependencies Required for E2E Tests
+
+It is likely that to complete E2E tests, a number of resources will be required as dependencies to enable the tests to pass successfully. Some examples:
+
+- When testing the Diagnostic Settings interface for a Resource Module, you will need an existing Log Analytics Workspace to be able to send the logs to as a destination.
+- When testing the Private Endpoints interface for a Resource Module, you will need an existing Virtual Network, Subnet and Private DNS Zone to be able to complete the Private Endpoint deployment and configuration.
+
+Module owners **MUST**:
+
+- Create the required resources that their module depends upon in the test file/directory
+  - They **MUST** use simple/native resource declarations/definitions in their respective IaC language
+  - They **MUST NOT** use other modules to deploy these required resources
```

Comment:
Not even published ones?
I'd recommend to reword this to "They must not use other local module references (...)" or something like that

Reply:
Nope we as we discussed yesterday on the office hours. Keeping tests supper simple and native declared is best as it's way easier to troubleshoot